code/modules/projectiles/ammunition/magazines.dm

```diff
@@ -258,6 +258,51 @@
 	caliber = "4.6x30mm"
 	max_ammo = 20
 	multi_sprite_step = 4
+	multiload = 0
+	w_class = WEIGHT_CLASS_NORMAL
+	///A var to check if the mag is being loaded
+	var/being_loaded = FALSE
+	/// There are two reloading processes ongoing so cancel them
+	var/double_loaded = FALSE
+
+/obj/item/ammo_box/magazine/wt550m9/attackby(obj/item/A, mob/user, params)
+	if(istype(A, /obj/item/ammo_box/wt550) || istype(A, /obj/item/ammo_box/magazine/wt550m9))
+		to_chat(user, "<span class='notice'>You begin to load the magazine with [src].</span>")
+		var/obj/item/ammo_box/AB = A
+		if(being_loaded == FALSE && double_loaded != TRUE)
+			being_loaded = TRUE
+			for(var/obj/item/ammo_casing/AC in AB.stored_ammo)
+				if(double_loaded == FALSE)// checks to make sure double_loaded hasn't been made true while an ongoing reload is already ongoing
+					if(length(stored_ammo) < max_ammo)
+						if(moving_do_after(user, 0.5 SECONDS, target = src, use_default_checks = TRUE))
+							src.give_round(AC)
+							AB.stored_ammo -= AC
+							update_mat_value()
+							update_appearance(UPDATE_DESC|UPDATE_ICON_STATE)
+							AB.update_appearance(UPDATE_DESC|UPDATE_ICON_STATE)
+							playsound(src, 'sound/weapons/gun_interactions/bulletinsert.ogg', 50, 1)
+						else
+							to_chat(user, "<span class='notice'>You stop loading the magazine with [src].</span>")
+							break
+					else
+						to_chat(user, "<span class='notice'>You stop loading the magazine with [src].</span>")
+						break
+				else
+					to_chat(user, "<span class='notice'>You stop loading the magazine with [src].</span>")
+					double_loaded = FALSE
+					break
+		else
+			double_loaded = TRUE //This will cancel ongoing reloads if made true by someone trying to do two reload processes at once
+			to_chat(user, "<span class='notice'>You're already loading the magazine!.</span>")
+			to_chat(user, "<span class='notice'>You stop loading the magazine with [src].</span>")
+		being_loaded = FALSE
+		return
+	if(istype(A, /obj/item/ammo_casing))
+		var/obj/item/ammo_casing/AC = A
+		if(give_round(AC))
+			user.drop_item()
+			AC.loc = src
+	return 0
```

Comment:
Please early return this proc at all oppertune moments, it might be worth moving lines 300-304 up to the start so returning after doesn't impact it

Reply:
ah right

Reply:
the breaks should cause an early return though, putting a return before them will lead to it just executing the loop again and putting a return after won't be executed.

Reply:
the breaks should cause early returns though

Reply:
An early return, or guard clause, would look like this:
```dm
if(!validity_check)
 return
do_stuff
```
Compared to what you currently have:
```dm
if(validity_check)
 do_stuff
```

Reply:
guard clauses added. Thanks for the advice about them I will try and use them in PRs going forward.